rl/agents/cem.py

```diff
@@ -0,0 +1,158 @@
+from __future__ import division
+from collections import deque
+from copy import deepcopy
+
+import numpy as np
+import keras.backend as K
+from keras.models import Model
+
+from rl.core import Agent
+from rl.util import *
+
+
+class CEMAgent(Agent):
+    def __init__(self, model, nb_actions, memory, window_length=1,
+                 gamma=.99, batch_size=50, nb_steps_warmup=1000, train_interval=50, elite_frac=0.05, memory_interval=1):
+
+        super(CEMAgent, self).__init__()
+
+        # Parameters.
+        self.nb_actions = nb_actions
+        self.gamma = gamma
+        self.batch_size = batch_size
+        self.elite_frac = elite_frac
+        self.num_best = int(self.batch_size*self.elite_frac)
+        self.nb_steps_warmup = nb_steps_warmup
+        self.train_interval = train_interval
+        self.memory_interval = memory_interval
+        self.window_length = window_length
+        self.episode=0
+
+        # Related objects.
+        self.memory = memory
+
+        self.model = model
+        self.shapes = [w.shape for w in model.get_weights()]
+        self.sizes = [w.size for w in model.get_weights()]
+        self.num_weights = sum(self.sizes)
+
+        self.update_theta(None)
+
+        # State.
+        self.compiled = False
+        self.reset_states()
+
+    def compile(self):
+        self.model.compile(optimizer='sgd', loss='mse')
+        self.compiled = True
+
+    def load_weights(self, filepath):
+        self.model.load_weights(filepath)
+        self.update_target_model_hard()
```

Comment:
Probably copy & paste problem. The call `self.update_target_model_hard()` should be removed since its not implemented here.